### PR TITLE
attach categories with product test

### PR DESCRIPTION
**Pull request by jayesh-parmar**

### CODE REVIEW

File: tests/Feature/ProductTest.php [hunk 1]
```diff
@@ -17,36 +22,44 @@
...
     ];
 
-    $this->post(route('product.store'), $productData)
-        ->assertStatus(302)
-        ->assertRedirect(route('product.index'));
-
-    $this->assertDatabaseHas('products', $productData);
+    $this->post(route('product.store'), array_merge($productData, ['category_ids' => $categoryIds]))
+    ->assertStatus(302)
+    ->assertRedirect(route('product.index'));
```
Comment (rajan-parmar) [inline tests/Feature/ProductTest.php]: instead of merging the categoryIds, add in the same array.

File: tests/Feature/ProductTest.php [hunk 2]
```diff
@@ -17,36 +22,44 @@
...
+    
+    $this->assertDatabaseHas('products', $productData);
```
Comment (rajan-parmar) [inline tests/Feature/ProductTest.php]: check category attach with product or not in the database.

**Review by rajan-parmar — CHANGES_REQUESTED** [review]

File: tests/Feature/ProductTest.php [hunk 3]
```diff
@@ -20,13 +20,28 @@
...
+        'status' => $productData['status'],
+    ]);
+
+        $product = Product::where('name', 'PDU')->first();
+        foreach ($categoryIds as $categoryId) {
+            $this->assertDatabaseHas('category_product', [
+                'product_id' => $product->id,
+                'category_id' => $categoryId,
+            ]);
+        }
```
Comment (rajan-parmar) [inline tests/Feature/ProductTest.php]: just check that one of the category_id is present. no need to check all ids

**Review by rajan-parmar — CHANGES_REQUESTED** [review]

File: tests/Feature/ProductTest.php [hunk 4]
```diff
@@ -81,13 +77,9 @@
...
-    $product = Product::where('name', 'PDU Update')->first();
-    foreach ($categoryIds as $categoryId) {
-        $this->assertDatabaseHas('category_product', [
-            'product_id' => $product->id,
-            'category_id' => $categoryId,
-        ]);
-    }
+        $product = Product::where('name', 'PDU Update')->first();
+        $this->assertTrue(
+              $product->categories->whereIn('id', $categoryIds)->isNotEmpty());
```
Comment (rajan-parmar) [inline tests/Feature/ProductTest.php]: use assertDatabaseHas but just check first category id only.

**Review by rajan-parmar — CHANGES_REQUESTED** [review]

File: tests/Feature/ProductTest.php [hunk 5]
```diff
@@ -77,9 +77,10 @@
         'status' => false,
     ] );
 
-        $product = Product::where('name', 'PDU Update')->first();
-        $this->assertTrue(
-              $product->categories->whereIn('id', $categoryIds)->isNotEmpty());
+    $firstCategoryId = reset($categoryIds); 
+    $this->assertDatabaseHas('category_product', [
+        'category_id' => $firstCategoryId,
+    ]);
```
Comment (rajan-parmar) [inline tests/Feature/ProductTest.php:82]: $categories = Category::factory(3)->create();

this give you the collection so play with this. passing only category Ids & same check first for database has id.

**Review by rajan-parmar — CHANGES_REQUESTED** [review]

File: tests/Feature/ProductTest.php [hunk 6]
```diff
@@ -9,45 +10,78 @@
 
     userLogin();
 
+    $categoryIds = Category::factory(3)->create()->pluck('id')->map(function ($id) {
+        return (string) $id;
+    })->toArray();
+
```
Comment (rajan-parmar) [inline tests/Feature/ProductTest.php]: map ni jarur nathi.

**Review by rajan-parmar — CHANGES_REQUESTED** [review]

File: tests/Feature/ProductTest.php [hunk 7]
```diff
@@ -10,9 +10,9 @@
 
     userLogin();
 
-    $categoryIds = Category::factory(3)->create()->pluck('id')->map(function ($id) {
-        return (string) $id;
-    })->toArray();
+    $categories = Category::factory(3)->create();
+    $string = implode(',', $categories->pluck('id')->toArray());
+    $categoryIds = explode(',', $string);
```
Comment (rajan-parmar) [inline tests/Feature/ProductTest.php:15]: why explode?
use https://laravel.com/docs/10.x/collections#method-implode

**Review by rajan-parmar — CHANGES_REQUESTED** [review]

File: tests/Feature/ProductTest.php [hunk 8]
```diff
@@ -9,45 +10,77 @@
 
     userLogin();
 
+    $categories = Category::factory(3)->create();
+    $string = implode(',', $categories->pluck('id')->toArray());
+    $categoryIds = explode(',', $string);
```
Comment (rajan-parmar) [inline tests/Feature/ProductTest.php]: why implode & explode required?

**Review by rajan-parmar — CHANGES_REQUESTED** [review]

File: tests/Feature/ProductTest.php (hunk 8 above)
Comment (jayesh-parmar) [reply tests/Feature/ProductTest.php]: category id ne string ma covert karva mate

File: tests/Feature/ProductTest.php [hunk 9]
```diff
@@ -9,45 +10,75 @@
...
+        'brand_id' => $productData['brand_id'],
+        'size_id' => $productData['size_id'],
+        'color_id' => $productData['color_id'],
+        'status' => $productData['status'],
+    ]);
 
-    $this->assertDatabaseHas('products', $productData);
+    $product = Product::where('name', 'PDU')->first();
+    $this->assertTrue(
+          $product->categories->whereIn('id', $categoryIds)->isNotEmpty());
```
Comment (rajan-parmar) [inline tests/Feature/ProductTest.php]: just check that one of the category id add in the pivot table.

**Review by rajan-parmar — CHANGES_REQUESTED** [review]

**Review by rajan-parmar — APPROVED** [review]

**Pull request merged**